app/jobs/regular/translate_posts.rb

```diff
@@ -28,7 +28,25 @@ def execute(args)
             .where.not(locale: nil)
             .where.not(locale: locale)
             .where("pl.id IS NULL")
-            .limit(limit)
+
+        if SiteSetting.automatic_translation_backfill_limit_to_public_content
+          public_categories = Category.where(read_restricted: false).pluck(:id)
+          posts =
+            posts
+              .joins(:topic)
+              .where(topics: { category_id: public_categories })
+              .where(topics: { archetype: "regular" })
```

Comment:
```suggestion
          posts =
            posts
              .joins(:topic)
              .where(topics: { category_id: Category.where(read_restricted: false).select(:id), archetype: "regular" })
```

We should avoid plucking all the ids into memory since it is not required and may be a problem on sites with alot of public categories.